### PR TITLE
Mimir query engine: update description for `-querier.enable-promql-engine-fallback` CLI flag

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR updates the description for the `-querier.enable-promql-engine-fallback` CLI flag to match the name change in https://github.com/grafana/mimir/pull/8291.

#### Which issue(s) this PR fixes or relates to

#8291

#### Checklist

- [n/a] Tests updated.
- [n/a] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [n/a] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: cmd/mimir/config-descriptor.json [hunk 1]
```diff
@@ -1924,7 +1924,7 @@
           "kind": "field",
           "name": "enable_promql_engine_fallback",
           "required": false,
-          "desc": "If set to true and the streaming engine is in use, fall back to using the Prometheus PromQL engine for any queries not supported by the streaming engine.",
+          "desc": "If set to true and the Mimir PromQL engine is in use, fall back to using the Prometheus PromQL engine for any queries not supported by the Mimir PromQL engine.",
```
Comment (jhesketh) [inline cmd/mimir/config-descriptor.json:1927]: I thought we were calling it "Mimir Query Engine"? It leaves open the possibility of extending it and we've been using the MQE acronym.

Comment (charleskorn) [reply cmd/mimir/config-descriptor.json:1927]: I was debating this - the flag name uses the term "PromQL engine", so I was trying to be consistent with that.

I'll change the term here.

**Review by jhesketh — APPROVED** [review]
Thanks :-). (FYI I haven't confirmed that you've found all instances)

**Comment by charleskorn** [comment]
> Thanks :-). (FYI I haven't confirmed that you've found all instances)

I did a quick search through the codebase, I couldn't find any more instances apart from the `streamingpromql` package name, which I don't want to touch while there are PRs in-flight.

**Review by tacole02 — COMMENTED** [review]
LGTM!

**Comment by tacole02** [comment]
Looks good!

**Pull request merged**